fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -427,6 +260,167 @@ def _threshold_optimization_equalized_odds(sensitive_features, labels, scores, g
+
+        X, y, sensitive_features = self._validate_input_data(
+            X, sensitive_features, y)
+
+        self.estimator_ = self.estimator.fit(X, y, **kwargs)
+
+        scores = _get_soft_predictions(self.estimator_, X)
+        self._post_processed_predictor_by_sensitive_feature = threshold_optimizer(
+            sensitive_features, y, scores, self.grid_size, self.flip, self.plot)
+
+    def predict(self, X, *, sensitive_features):
+        """Predict label for each sample in X while taking into account sensitive features.
+
+        :param X: feature matrix
+        :type X: numpy.ndarray or pandas.DataFrame
+
+        :param sensitive_features: sensitive features to identify groups by, currently allows
+            only a single column
+        :type sensitive_features: currently 1D array as numpy.ndarray, list, pandas.DataFrame,
+            or pandas.Series
+
+        :return: predictions in numpy.ndarray
+        """
+        check_is_fitted(self)
+        random_state = check_random_state(self.random_state)
+
+        X, _, sensitive_features = self._validate_input_data(
+            X, sensitive_features)
+
+        unconstrained_predictions = self._unconstrained_predictor.predict(X)
+
+        positive_probs = _vectorized_prediction(
+            self._post_processed_predictor_by_sensitive_feature,
+            sensitive_features,
+            unconstrained_predictions)
+        return (positive_probs >= random_state.rand(len(positive_probs))) * 1
+
+    def _pmf_predict(self, X, *, sensitive_features):
+        """Probabilistic mass function.
+
+        :param X: Feature matrix
+        :type X: numpy.ndarray or pandas.DataFrame
+
+        :param sensitive_features: Sensitive features to identify groups by,
+            currently allows only a single column
+
+        :type sensitive_features: Currently 1D array as numpy.ndarray, list,
+            pandas.DataFrame, or pandas.Series
+
+        :return: array of tuples with probabilities for predicting 0 or 1,
+            respectively. The sum of the two numbers in each tuple needs to add
+            up to 1.
+        :rtype: numpy.ndarray
+        """
+        check_is_fitted(self)
+        X, _, sensitive_features = self._validate_input_data(
+            X, sensitive_features)
+        positive_probs = _vectorized_prediction(
+            self._post_processed_predictor_by_sensitive_feature, sensitive_features,
+            self._unconstrained_predictor.predict(X))
+        return np.array([[1.0 - p, p] for p in positive_probs])
+
+    def _validate_input_data(self, X, sensitive_features, y=None):
+        if y:
+            X, y = check_X_y(X, y)
+            y = check_array(y, ensure_2d=False, dtype=int)
+        else:
+            X = check_array(X)
+        check_consistent_length(X, sensitive_features)
+
+        if set(np.unique(y)) > set([0, 1]):
+            raise ValueError(NON_BINARY_LABELS_ERROR_MESSAGE)
```

Comment:
this should be `not (set(np.unique(y)) <= set([0, 1])`.

Reply:
I think this was just copied from the existing validation which I am working through at the moment myself, and it seems off. E.g. the labels [1,2] are not ">" than [0,1] because that's superset. What it should be IMO
```
not set(np.unique(y)).issubset(set([0, 1]))
```
which should be identical to yours, Miro, but I find that way of writing more intuitive

Reply:
I don't have a strong preference between `<=` and `issubset`.

Reply:
yeah this is a copy/paste from existing code, we can fix it in a different PR. I think there are better ways to handle the [categorical] output anyway. Just didn't wanna tackle that in this PR.

Reply:
I believe I'm fixing this in my PR so feel free to close this thread